# Importing the libraries :
***

In [20]:
import pandas as pd
import numpy as np

## Importing thr data into the dataframe :
***

In [11]:
df = pd.read_csv("ECG_training.csv")
df

ECG Classification
0     -127,-162,-197,-229,-245,-254,-261,-265,-268,-...              N
1     128,157,189,226,250,257,262,265,268,269,268,26...              N
2     519,619,723,827,914,956,955,934,920,900,889,88...              A
3     -266,-316,-367,-407,-423,-423,-401,-367,-329,-...              N
4     21,22,24,26,28,31,32,34,34,35,35,34,34,34,35,3...              N
...                                                 ...            ...
3495  347,378,410,540,695,736,751,778,849,803,688,67...              N
3496  -59,-71,-87,-104,-111,-114,-117,-119,-120,-121...              N
3497  -60,-83,-106,-128,-145,-155,-163,-168,-173,-17...              N
3498  582,702,807,893,962,997,984,945,927,915,906,89...              N
3499  162,145,97,27,-36,-64,-90,-89,-37,61,188,324,4...              N

[3500 rows x 2 columns]

## Splitting the data in the ECG column into different columns :
***

In [12]:
new_df = df.join(df['ECG'].str.split(',',expand = True))
df_final=new_df.drop('ECG',axis=1)
df_final

Classification     0     1     2     3     4     5     6     7     8  \
0                 N  -127  -162  -197  -229  -245  -254  -261  -265  -268   
1                 N   128   157   189   226   250   257   262   265   268   
2                 A   519   619   723   827   914   956   955   934   920   
3                 N  -266  -316  -367  -407  -423  -423  -401  -367  -329   
4                 N    21    22    24    26    28    31    32    34    34   
...             ...   ...   ...   ...   ...   ...   ...   ...   ...   ...   
3495              N   347   378   410   540   695   736   751   778   849   
3496              N   -59   -71   -87  -104  -111  -114  -117  -119  -120   
3497              N   -60   -83  -106  -128  -145  -155  -163  -168  -173   
3498              N   582   702   807   893   962   997   984   945   927   
3499              N   162   145    97    27   -36   -64   -90   -89   -37   

      ...  8990  8991  8992  8993  8994  8995  8996 8997 8998 8999  
0     ...    36    19     8     0    -7   -12   -15  -18  -22  -21  
1     ...    -7    -5    -3    -2    -1    -1     0    0    1    2  
2     ...  1142  1144  1055   866   632   403   224  116   17   18  
3     ...    76    74    73    69    68    66    62   51   34   21  
4     ...  -634  -456  -263   -46   133   227   257  236  174   84  
...   ...   ...   ...   ...   ...   ...   ...   ...  ...  ...  ...  
3495  ...   229   241   267   252   115    10     1   37   26  -80  
3496  ...   152   120    92    63    32    16     7    2    0   -4  
3497  ...    57    52    47    42    34    18     5   -5   -8   -9  
3498  ...   -26   -25   -23   -13   -38   -65   -65  -50  -42  -46  
3499  ...  -233  -226  -217  -196  -157  -128  -104  -76  -51  -30  

[3500 rows x 9001 columns]

## Splitting the data into training and testing sets :
***

In [13]:
X= df_final.drop('Classification',axis=1)
y=df_final['Classification']

In [14]:
from sklearn.model_selection import train_test_split

X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2)


## Fitting the data into the classification models :
***

In [22]:
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn import svm

models = {"Logistic Regression": LogisticRegression(max_iter=100),
          "KNN": KNeighborsClassifier(),
          "Random Forest": RandomForestClassifier(n_estimators=100),
          "Decision Tree": DecisionTreeClassifier(),
          " SVM :": svm.SVC()}

# Create a function to fit and score models
def fit_and_score(models, X_train, X_test, y_train, y_test):
   
    np.random.seed(42)
   
    model_scores = {}
   
    for name, model in models.items():
       
        model.fit(X_train, y_train)
       
        model_scores[name] = model.score(X_test, y_test)
    return model_scores

In [23]:
model_scores = fit_and_score(models=models,
                             X_train=X_train,
                             X_test=X_test,
                             y_train=y_train,
                             y_test=y_test)

model_scores

C:\Users\pjp\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


{'Logistic Regression': 0.5328571428571428,
 'KNN': 0.8714285714285714,
 'Random Forest': 0.8714285714285714,
 'Decision Tree': 0.7685714285714286,
 ' SVM :': 0.8742857142857143}

# As you can see.....SVM,Random Forests and KNN gives us the maximum efficiency of almost around 87%